app/services/service-worker.js

```diff
@@ -77,6 +79,19 @@ export default Service.extend(Evented, {
 		});
 	},
 
+	/*
+	 * Delete caches
+	 */
+	purgeCache() {
+		const _caches = this.get('caches');
+
+		return _caches.keys().then((cacheNames = []) => {
```

Comment:
Delete every cache in the site is not a good thing, you could be deleting other caches not related with SW's. Can we only delete caches created by SW's?

Reply:
Sadly, I don't think we can do that, as we don't have any relationship between serviceworker and the cache names he's using. What we can have with a simple change is a boolean flag as the fulfilment value of `unregisterAll` return promise, that will tell us if the method unregistered something or it was a no-op. Would that work for you?